lib/connect_proto/src/substance.proto

```diff
@@ -0,0 +1,32 @@
+syntax = "proto3";
+package primary.connect;
+option go_package = "primary.connect";
+
+import "google/protobuf/timestamp.proto";
+
+import "coded_value.proto";
+
+message Substance {
+  message Product {
+    CodedValue code = 1; // Code for the result component
+    string lot_number = 2;
+    google.protobuf.Timestamp expiration_date = 3;
+    CodedValue manufacturer = 4;
+  }
+
+  message Route {
+    CodedValue code = 1; // Code for the result component
+  }
+
+  message Site {
+    CodedValue code = 1; // Code for the result component
+  }
+
+  message Dose {
+    string quantity = 1;
```

Comment:
Would we also want to track which dose it is? For example, we had two initial Covid19 vaccines, followed by a bootster. Do we need to encode that information into our model?

Reply:
Good idea, how about a series field?